apsuite/optimization/rcds.py

```diff
@@ -431,15 +454,186 @@ def _optimize(self):
+        ax.set_xlabel("evaluations")
+
+        ax.legend()
+
+        ax = axs[1]
+        colors = _mplt.rcParams["axes.prop_cycle"].by_key()["color"]
+
+        for i, knob in enumerate(_np.array(self.positions_evaluated).T):
+            color = colors[i % len(colors)]
+
+            ax.plot(knob, alpha=0.4, color=color)
+            ax.plot(
+                opt_idcs,
+                pos_cum_opt[:, i],
+                "o", mfc="none",
+                color=color,
+                label=f"dir {i + 1:2d} - cumulated optima"
+
+            )
+            ax.plot(
+                iters_idcs,
+                self.positions_best[:, i], 'o',
+                color=color,
+                label=f"dir {i + 1:2d} - end of iter. optima"
+                )
+
+        ax.set_ylabel("knobs values")
+        ax.set_xlabel("evaluations")
+        ax.legend()
+
+        fig.tight_layout()
+        fig.show()
+
+        return fig, axs
+
+    def plot_knobspace_slice(self, dir_idcs=(1, 2)):
+        """Plot slice of parameter space (knobs space).
+
+        Args:
+            dir_idcs (tuple, list): Indices of the desired knobs directions
+            (eg. (1,2), (1,3) etc.) . Defaults to (1, 2).
+
+        Returns:
+            fig, ax: matplotlib fig and ax
+        """
+        idx1, idx2 = dir_idcs
+        idx1 -= 1
+        idx2 -= 1
+
+        pos_eval = _np.array(self.positions_evaluated)
+        pos_best = _np.array(self.positions_best)
+        objfuncs_eval = _np.array(self.objfuncs_evaluated)
+        _, pos_cum_opt, _ = self.get_cumulated_optimum()
+
+        knob1_eval = pos_eval[:, idx1]
+        knob2_eval = pos_eval[:, idx2]
+
+        knob1_cum_opt = pos_cum_opt[:, idx1]
+        knob2_cum_opt = pos_cum_opt[:, idx2]
+
+        knob1_enditer = pos_best[:, idx1]
+        knob2_enditer = pos_best[:, idx2]
+
+        fig, ax = _mplt.subplots()
+        scatter = ax.scatter(
+            x=knob1_eval, y=knob2_eval,
+            c=objfuncs_eval,
+            vmin=objfuncs_eval.min(),
+            vmax=objfuncs_eval.max(),
+            label="positions evaluated"
+        )
+
+        ax.plot(
+            knob1_cum_opt,
+            knob2_cum_opt,
+            "o", color="red", mfc="none",
+            label="cumulated optima"
+        )
+
+        ax.plot(
+            knob1_enditer,
+            knob2_enditer,
+            "x", color="red",
+            label="end of iter. optima"
+        )
+
+        ax.plot(
+            knob1_enditer[-1],
+            knob2_enditer[-1],
+            "d", color="magenta",
+            markersize=10,
+            label="end of run optimum"
+        )
+
+        ax.set_xlabel(f"dir {idx1 + 1}")
+        ax.set_ylabel(f"dir {idx2 + 1}")
+
+        colorbar = fig.colorbar(scatter, ax=ax)
+        colorbar.set_label('objective function value')
+
+        ax.legend()
+        ax.set_title("parameter space slice")
+
+        return fig, ax
+
+    def get_cumulated_optimum(self):
```

Comment:
why isn't this method at the base class?

Reply:
With the generalization, now it is.